sdk/servicebus/service-bus/src/util/errors.ts

```diff
@@ -249,3 +249,27 @@ export function throwErrorIfInvalidOperationOnMessage(
     throw error;
   }
 }
+
+/**
+ * Error message for when the ServiceBusMessage provided by the user has different values
+ * for partitionKey and sessionId.
+ * @internal
+ * @throw
+ */
+export const PartitionKeySessionIdMismatchError =
+  "The fields 'partitionKey' and 'sessionId' cannot have different values.";
```

Comment:
Changelog

Reply:
I'll add one in a separate PR with the other validation checks I added in previous PRs